docs/book/src/migration/v1vsv2.md

```diff
@@ -0,0 +1,61 @@
+# Kubebuilder v1 vs v2
+
+There are a number of changes in
+[controller-runtime](https://github.com/kubernetes-sigs/controller-runtime),
+[controller-tools](https://github.com/kubernetes-sigs/controller-tools)
+and [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder).
+
+## Common changes
+
+- They have migrated from `dep` to `go modules` for managing dependencies.
+
+## Controller-runtime
+
+- Automatic certificate generation for webhooks has been removed, and webhooks
+will no longer self-register. Use controller-tools to generate a webhook
+configuration, and use something like cert-manager to generate your certificates
+(if you don't already have a solution). Kubebuilder supports them out-of-the-box.
+
+- If your API implements
+[`webhook.Defaulter`](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook#Defaulter)
+and [`webhook.Validator`](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook#Validator),
+a webhook server will be automatically wired for you.
+The tutorial can be found at [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+
+- Add CRD conversion webhook support. The tutorial can be found [here](https://github.com/kubernetes-sigs/kubebuilder/pull/851).
+
+- `manager` now supports adding runables that is either managed by leader election
+or always running. It depends on how runable implements
+[LeaderElectionRunable](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#LeaderElectionRunnable)
+interface.
+
+- `client` package:
+  - `client.List` now uses functional options.
+  - Support delete collection.
```

Comment:
Lang: this construction is awkward (also, perhaps confusingly, a singluar thing `supports`, plural things `support`)